src/Query/Builder.php

```diff
@@ -659,6 +660,26 @@ public function whereBetween($column, iterable $values, $boolean = 'and', $not =
 
     /** @inheritdoc */
     public function insert(array $values)
+    {
+        return $this->performInsert($values);
+    }
+
+    /** @inheritdoc */
+    public function insertOrIgnore(array $values)
+    {
+        try {
+            return $this->performInsert($values, ['ordered' => false]);
+        } catch (BulkWriteException $exception) {
+            // Ignore "duplicate key error"
+            if ($exception->getCode() !== 11000) {
```

Comment:
I'm not sure if I should ignore only "duplicate key errors". [Laravel docs](https://laravel.com/docs/11.x/queries#insert-statements)
> The `insertOrIgnore` method will ignore errors while inserting records into the database. When using this method, you should be aware that duplicate record errors will be ignored and other types of errors may also be ignored depending on the database engine.

Reply:
I dislike the ambiguity that is built into the documentation - you can't really be sure what's ignored. One option is to ignore any errors that purely come from the data (e.g. unique key violations, document constraint violations) but still report more serious problems (like server selection errors). IMO, users should be more aware of what could go wrong and explicitly ignore those errors if they don't care for them.

For the purpose of the cache/lock work, this seems sufficient so I'm fine with only ignoring unique key errors and returning to this if people have more errors they think should be ignored.

Reply:
This method is no longer used by the cache. I'm reverting this change, and creating a dedicated ticket PHPORM-170